pyuavcan/transport/can/media/pythoncan/_pythoncan.py

```diff
@@ -26,25 +26,64 @@ class PythonCANMedia(_media.Media):
     A media interface adapter for `python-can <https://github.com/hardbyte/python-can>`_.
 
     - Usage example for PCAN-USB channel 1 (bitrate = 500k, mtu = 8, Node-ID = 10)::
-
-        --tr="CAN(can.media.pythoncan.PythonCANMedia('pcan','PCAN_USBBUS1',5000000,8),10)"
-
+        --tr="CAN(can.media.pythoncan.PythonCANMedia('pcan:PCAN_USBBUS1',5000000,8),10)"
+    - Usage example for PCAN-USB channel 1 (nom.bitrate = 500k, data.bitrate = 2M, mtu = 8, Node-ID = 10)::
+        --tr="CAN(can.media.pythoncan.PythonCANMedia('pcan:PCAN_USBBUS1',[5000000,2000000],8),10)"
     - Usage example for Kvaser channel 0 (bitrate = 500k, mtu = 8, Node-ID = 10)::
-
-        --tr="CAN(can.media.pythoncan.PythonCANMedia('kvaser','0',5000000,8),10)"
+        --tr="CAN(can.media.pythoncan.PythonCANMedia('kvaser:0',5000000,8),10)"
     """
 
-    def __init__(self, iface_name: str, channel_name: str, bitrate: int, mtu: int) -> None:
+    VALID_FD_ADAPTER_SET = {'pcan', 'kvaser'}
+
+    def __init__(self, iface_name: str, bitrate: typing.Union[int, typing.Tuple[int, int]], mtu: int) -> None:
         self._iface_name = str(iface_name)
-        self._channel_name = str(channel_name)
+        self._conn_name = self._iface_name.split(':')
+        if len(self._conn_name) != 2:
+            raise RuntimeError('Wrong interface name format')
+        if mtu not in self.VALID_MTU_SET:
+            raise RuntimeError('Wrong MTU value: {}'.format(mtu))
         self._mtu = int(mtu)
-        self._is_fd = self._mtu > _NativeFrameDataCapacity.CAN_CLASSIC
+        self._is_fd = self._mtu > min(self.VALID_MTU_SET) or type(bitrate) != int
         self._loop = asyncio.get_event_loop()
         self._closed = False
         self._maybe_thread: typing.Optional[threading.Thread] = None
         self._background_executor = concurrent.futures.ThreadPoolExecutor(max_workers=3)
         self._loopback_enabled = False
-        self._bus = can.ThreadSafeBus(interface=iface_name, channel=channel_name, bitrate=bitrate)
+        if not isinstance(bitrate, int):
+            nom_br = int(bitrate[0])
+            data_br = int(bitrate[1])
+        else:
+            nom_br = int(bitrate)
+            data_br = int(bitrate)
+        if self._is_fd == False:
+            #
+            self._bus = can.ThreadSafeBus(interface=self._conn_name[0], channel=self._conn_name[1], bitrate=nom_br)
+        elif self._conn_name[0] == 'pcan':
```

Comment:
This is not to pontificate but this design suffers from the lack of orthogonality. You have your CAN FD-capable adapters listed in `VALID_FD_ADAPTER_SET` and then you make assumptions about the contents of that set in this long `elif`. A cleaner, orthogonal design would keep pertinent elements in one place only and avoid unnecessary logical coupling. You can attain this by replacing your `VALID_FD_ADAPTER_SET` with a dict that maps the interface name (like `pcan`) to a static factory function. Then in the constructor you go like:

```python
try:
    self._bus = self.FD_ADAPTERS[self._conn_name[0]](things...)
except KeyError:
    raise can't use CAN FD on this iface
```

Reply:
The problem is I don't know how to convert integer or tuple of integers properly without getting any error messages from mypy. If I don't check the type of baudrate with instance, mypy doesn't want to accept int().

Reply:
And actually I don't use VALID_FD_ADAPTER_SET now